### PR TITLE
kernel-install: cleanups

### PR DESCRIPTION
**Pull request by yuwata**
split-out of #26848.

### CODE REVIEW

File: src/kernel-install/kernel-install.in [hunk 1]
```diff
@@ -217,15 +217,15 @@ for pref in $BOOT_ROOT_SEARCH; do
...
-        fi
     done
+
+    if [ -d "$pref/loader/entries" ]; then
+        [ -z "$BOOT_ROOT" ] && BOOT_ROOT="$pref"
+        log_verbose "$pref/loader/entries exists, using BOOT_ROOT=$BOOT_ROOT"
+        break
+    else
+        log_verbose "$pref/loader/entries not found…"
+    fi
```
Comment (keszybz) [inline src/kernel-install/kernel-install.in:228]: Actually this *does* change behaviour, for the better. If we have multiple entries in `$ENTRY_TOKEN_SEARCH`, and `$pref/loader/entries` exists, we would always exit after the first inner loop and ignore the second and later candidates.

**Review by keszybz — COMMENTED** [review]
Yeah, let's merge this.

For the first commit, it could still be made to work, by more carefully checking only for partitions on the disk on which root is. This would be great because it'd completely free us from depending on stuff being mounted in the right place. You could even mount ESP on `/tmp/esp` and XBOOTLDR on `/tmp/xboot` and things have magically work. But let's revert now, as the issue is not easy to fix.

File: src/kernel-install/kernel-install.in (hunk 1 above)
Comment (keszybz) [reply src/kernel-install/kernel-install.in:228]: So… please just adjust the commit message.

Comment (yuwata) [reply src/kernel-install/kernel-install.in:228]: Indeed. I did not notice that.

**Pull request merged**